### PR TITLE
feat:Session6 using Github API

### PR DESCRIPTION
**Pull request by gavandivya**
image
image
image
Fetched 10 Github User profiles using GitHub API and Implemented Search Functionality on it.
Please give your valuable feedback. It will help a lot.

### CODE REVIEW

File: Lecture6(GithubAPI)/Component/CardComponent.js [hunk 1]
```diff
@@ -10,8 +10,8 @@ const CardComponent = ({ details }) => {
     avatar_url,
   } = details;
   return (
-    <div className="col-sm-6 col-md-4 col-lg-3 col-xl-3 mt-4">
-      <div className="card" style={{ width: "11rem" }}>
+    <div className="col-sm-6 col-md-4 col-lg-3 col-xl mt-4">
+      <div className="card">
         <img src={avatar_url} className="card-img-top img-fluid" alt="" />
```
Comment (Pujarini) [inline Lecture6(GithubAPI)/Component/CardComponent.js]: give a alt text. It is good for accessibility.

File: Lecture6(GithubAPI)/Component/BodyComponent.js [hunk 2]
```diff
@@ -26,14 +28,13 @@ const BodyComponent = () => {
     useEffect(() => {
         if (!sessionStorage.getItem('teamData')) {
             getGithubApiData().then((githubdata) => {
```
Comment (Pujarini) [inline Lecture6(GithubAPI)/Component/BodyComponent.js]: Are you trying data from session storage and api if not available in session storage? 
session storages expires as soon as a tab is closed.
Also what if you get an error from the api you are fetching data from?

Comment (Pujarini) [reply Lecture6(GithubAPI)/Component/BodyComponent.js]: I would suggest local storage for that. Also now you can handle api cases.

File: Lecture6(GithubAPI)/Component/SearchComponent.js [hunk 3]
```diff
@@ -23,7 +23,7 @@ const SearchComponent = ({ DataFromAPI, setFilteredData, setNoRecord }) => {
       onSubmit={(e) => {
         e.preventDefault();
         const filteredData = searchTeamdetails(searchText, DataFromAPI);
-        console.info(filteredData);
+        // console.info(filteredData);
         if (filteredData.length > 0) {
           setFilteredData(filteredData);
```
Comment (Pujarini) [inline Lecture6(GithubAPI)/Component/SearchComponent.js:27]: What would be value of filteredData (as in state) then when filteredData length is 0?

**Comment by Pujarini** [comment]
great work Divya on UI front! 🎉 I have added some comments you can check them out.

**Review by Pujarini — CHANGES_REQUESTED** [review]

File: Lecture6(GithubAPI)/Component/BodyComponent.js (hunk 2 above)
Comment (gavandivya) [reply Lecture6(GithubAPI)/Component/BodyComponent.js]: Yes will implement error handling

**Comment by nikitaj-57** [comment]
Reviewed and it looks good. Just one suggestion - you can add NoRecord component in the components folder just like the header component.

File: Lecture6(GithubAPI)/Component/CardComponent.js (hunk 1 above)
Comment (gavandivya) [reply Lecture6(GithubAPI)/Component/CardComponent.js]: done

**Comment by gavandivya** [comment]
> great work Divya on UI front! 🎉 I have added some comments you can check them out.

Thanks, bro! sure will check it out

**Comment by gavandivya** [comment]
> Reviewed and it looks good. Just one suggestion - you can add NoRecord component in the components folder just like the header component.

Sure will do it!

File: Lecture6(GithubAPI)/Component/SearchComponent.js (hunk 3 above)
Comment (gavandivya) [reply Lecture6(GithubAPI)/Component/SearchComponent.js:27]: it will be [] empty array when searched with anything which don't matches.

**Comment by roshantrivedi** [comment]
Reviewed. Nice work here.

File: Lecture4/Component/CardComponent.js [hunk 4]
```diff
@@ -9,7 +9,7 @@ const CardComponent = ({ details }) => {
         <img
           src={profilePic}
           className="card-img-top img-fluid"
-          alt=""
+          alt="profilePic"
```
Comment (ap221882) [inline Lecture4/Component/CardComponent.js:12]: screen readers will read it as profilePic's image. Can change it to {name}.

File: Lecture4/Component/HeaderComponent.js [hunk 5]
```diff
@@ -1,10 +1,11 @@
+import { SocialMediaIcon as icon } from "../socialMediaIcon";
```
Comment (ap221882) [inline Lecture4/Component/HeaderComponent.js:1]: Leave a blank line after imports.

File: Lecture4/Component/HeaderComponent.js [hunk 6]
```diff
@@ -1,10 +1,11 @@
...
 const HeaderComponent = () => (<nav className="navbar navbar-light bg-light">
     <a className="navbar-brand" href="#">
         <img
             className="d-inline-block align-top mr-4"
             width={50}
```
Comment (ap221882) [inline Lecture4/Component/HeaderComponent.js:6]: Avoid inline and html attribute styling. Decreases code readability.

File: Lecture6(GithubAPI)/Component/BodyComponent.js [hunk 7]
```diff
@@ -0,0 +1,84 @@
+import { useState, useEffect } from "react";
+import Skeleton from 'react-loading-skeleton'
+import 'react-loading-skeleton/dist/skeleton.css'
+import { SkeletonTheme } from "react-loading-skeleton";
+
```
Comment (ap221882) [inline Lecture6(GithubAPI)/Component/BodyComponent.js:5]: Appreciated! was readable

File: Lecture6(GithubAPI)/Component/BodyComponent.js [hunk 8]
```diff
@@ -0,0 +1,84 @@
...
+        if (!sessionStorage.getItem('teamData')) {
+            getGithubApiData().then((githubdata) => setDataFromAPI(githubdata))
+                .catch((err) => {
+                    console.log(err);
+                    setNoRecord(true);
+                })
+                .finally(() => setLoading(false));
+        }
+        else {
+            setDataFromAPI(JSON.parse(sessionStorage.getItem('teamData')));
```
Comment (ap221882) [inline Lecture6(GithubAPI)/Component/BodyComponent.js:41]: Very nice. Will add this to my code too. I was exhausting the API requests every time.

**Review by ap221882 — COMMENTED** [review]
It was well written. Kudos Divya!

**Pull request merged**